src/bindgen/graph/struct.cr

```diff
@@ -1,21 +1,25 @@
 module Bindgen
   module Graph
-    # A `struct` in Crystal (`lib` or not), or a plain `struct` in C++.
+    # A `struct` in Crystal (`lib` or not), or a plain `struct` in C++.  Also
+    # represents a `lib union`.
     #
     # A structure can host both raw variable fields (C-style) and other methods
     # at once.
     class Struct < Container
       # Used to signal the `Generator::Cpp` to generate a `using BASE::BASE;`.
       INHERIT_CONSTRUCTORS_TAG = "INHERIT_CONSTRUCTORS_TAG"
 
-      # Fields in this struct.
+      # Fields in this structure.
       getter fields : Hash(String, Call::Result)
 
       # Name of the base-class, if any.  This is mainly useful for C++ to
       # generate the jump-table.
       property base_class : String?
 
-      def initialize(@fields, name, parent, @base_class = nil)
+      # Does this structure represent a C union?
+      getter? c_union : Bool
```

Comment:
In C land, for some reason, unions and structs are the same - Except for their memory layout.  I think having a proper `class Union < Container` (Or a different name as `Union` itself collides with Crystal -.-;) would be better.  Also as at least `@base_class` isn't applicable to unions.